File: scaffold/src/main/java/top/xuqingquan/web/publics/DefaultWebLifeCycleImpl.java
package top.xuqingquan.web.publics;

import top.xuqingquan.web.nokernel.WebConfig;
import top.xuqingquan.web.nokernel.WebLifeCycle;

public class DefaultWebLifeCycleImpl implements WebLifeCycle {
    private android.webkit.WebView mWebView;
    private com.tencent.smtt.sdk.WebView mX5WebView;

    public DefaultWebLifeCycleImpl(android.webkit.WebView webView) {
        this.mWebView = webView;
    }

    public DefaultWebLifeCycleImpl(com.tencent.smtt.sdk.WebView webView) {
        this.mX5WebView = webView;
    }

    @Override
    public void onResume() {
        if (WebConfig.hasX5()) {
            if (this.mX5WebView != null) {
                this.mX5WebView.onResume();
                this.mX5WebView.resumeTimers();
            }
        } else {
            if (this.mWebView != null) {
                this.mWebView.onResume();
                this.mWebView.resumeTimers();
            }
        }
    }

    @Override
    public void onPause() {
        if (WebConfig.hasX5()) {
            if (this.mX5WebView != null) {
                this.mX5WebView.onPause();
                this.mX5WebView.pauseTimers();
            }
        } else {
            if (this.mWebView != null) {
                this.mWebView.onPause();
                this.mWebView.pauseTimers();
            }
        }
    }

    @Override
    public void onDestroy() {
        if (WebConfig.hasX5()) {
            if (this.mX5WebView != null) {
                this.mX5WebView.resumeTimers();
            }
            AgentWebUtils.clearWebView(this.mX5WebView);
        } else {
            if (this.mWebView != null) {
                this.mWebView.resumeTimers();
            }
            AgentWebUtils.clearWebView(this.mWebView);
        }
    }
}

File: scaffold/src/main/java/top/xuqingquan/webbak/publics/DefaultWebLifeCycleImpl.kt
package top.xuqingquan.webbak.publics

import top.xuqingquan.webbak.nokernel.WebConfig
import top.xuqingquan.webbak.nokernel.WebLifeCycle

class DefaultWebLifeCycleImpl : WebLifeCycle {
    private var mWebView: android.webkit.WebView? = null
    private var mX5WebView: com.tencent.smtt.sdk.WebView? = null

    constructor(webView: android.webkit.WebView?) {
        this.mWebView = webView
    }

    constructor(webView: com.tencent.smtt.sdk.WebView?) {
        this.mX5WebView = webView
    }

    override fun onResume() {
        if (WebConfig.hasX5()) {
            mX5WebView?.let {
                it.onResume()
                it.resumeTimers()
            }
        } else {
            mWebView?.let {
                it.onResume()
                it.resumeTimers()
            }
        }
    }

    override fun onPause() {
        if (WebConfig.hasX5()) {
            mX5WebView?.let {
                it.onPause()
                it.pauseTimers()
            }
        } else {
            mWebView?.let {
                it.onPause()
                it.pauseTimers()
            }
        }
    }

    override fun onDestroy() {
        if (WebConfig.hasX5()) {
            this.mX5WebView?.resumeTimers()
            AgentWebUtils.clearWebView(this.mX5WebView)
        } else {
            this.mWebView?.resumeTimers()
            AgentWebUtils.clearWebView(this.mWebView)
        }
    }
}
